django-rgd/rgd/models/file.py

```diff
@@ -103,6 +103,10 @@ class Meta:
     def basename(self):
         return os.path.basename(self.name)
 
+    @property
+    def size(self):
+        return self.file.size
```

Comment:
This will likely raise an error for URL files as `self.file` should be `None`
```suggestion
    @property
    def size(self):
        if self.type == FileSourceType.FILE_FIELD:
            return self.file.size
        # NOTE: we do not have an easy way of getting URL file sizes on the fly
```

Reply:
Good catch! I forgot about that case, I'm not exactly sure what to do here. It seems to me the two obvious options are:
1. If it's a URL file, return a size of zero. The problem with this is that now we're conveying an incorrect size field, without any other indication that we're unaware of the actual size.
2. If it's a URL file, mark its size, and any folder that contains it, as `unknown`. This would apply if any folder has any URL files, converting its size to `unknown`. The benefit of this is that we're not confidently displaying an incorrect size, but the detriment is that we're also not conveying _any_ information, even if we do have some file fields.

Thinking about this now, there may be a third approach, which would bridge the gap between these two options. We could change the `size` field on folders to `min_size`, to capture any size info on normal file fields. We could also add a `num_url_files` field to the response, to indicate how many files are missing from the `min_size` field. If `num_url_files == 0`, then `min_size` is the actual size of the folder.

@banesullivan Any thoughts on this?

Reply:
I like the `unknown` value for URL files and using `min_size` to represent the knowledge gap if a folder contains a mix between URL files and FieldFiles